### ISSUE TITLE
AMP Loss -> Nan

### ISSUE THREAD

**Issue by ZhikangNiu** [NoFish-528/encodec-pytorch#8]
the amp training isn't stable. The loss will be Nan

**Comment by ZhikangNiu** [comment]
I found this bug is caused by the [laplace_smoothing](https://github.com/ZhikangNiu/encodec-pytorch/blob/401041daddc2ca82f88aacbd95f120cc52620dd0/quantization/core_vq.py#L59)
We can scale the epsilon to 1e-3 and reduce learning rate. It can work well by using fp16 !

**Comment by ZhikangNiu** [comment]
![image](https://github.com/ZhikangNiu/encodec-pytorch/assets/73390819/8ce8e4b9-0274-4eee-bb97-a4438f785ced)

**Comment by SaladDay** [comment]
Hello, I have a doubt, why does the updated Nan in VQ affect the overall loss function?

**Comment by ZhikangNiu** [comment]
> Hello, I have a doubt, why does the updated Nan in VQ affect the overall loss function?

It will cause task failure
